xc/domain.ml

```diff
@@ -230,15 +235,25 @@ let make ~xc ~xs vm_info uuid =
 				let ent = sprintf "%s/%s" dom_path dir in
 				t.Xst.mkdirperms ent roperm
 			) [ "cpu"; "memory" ];
+
+			let mksubdirs base dirs perms =
```

Comment:
I don't think you're missing anything. Essentially I modelled the new `mksubdirs` on the preexisting code, in which the `List.map` was and is to minimise repetitions of the string "device".

We want to include make the subnodes under "device" in each of our two calls to `mksubdirs`, hence my including that code-snippet in the function itself. Still, it does end up harder to read than it could be. I'll improve it.

`Filename.concat` would not be appropriate semantically, even though it would happen to do the right thing by coincidence because we will be running in a Unix-like environment where the directory separator matches xenstore's node-separator character.